compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -898,34 +1245,53 @@ fn transform_ty<'tcx>(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, options: TransformTyOptio
             }
         }
 
-        ty::FnDef(def_id, args) => {
-            ty = Ty::new_fn_def(tcx, *def_id, transform_args(tcx, args, options));
+        ty::FnDef(def_id, ..) => {
+            // Transform function items into function pointers
+            let fn_sig = tcx.fn_sig(def_id).skip_binder();
+            let fn_ptr = Ty::new_fn_ptr(tcx, fn_sig);
+            // Transform fn_sig inputs and output
+            ty = transform_ty(tcx, fn_ptr, options);
         }
 
-        ty::Closure(def_id, args) => {
-            ty = Ty::new_closure(tcx, *def_id, transform_args(tcx, args, options));
+        ty::Closure(_, args) => {
```

Comment:
This seems to just be totally dropping upvars on the floor?

Reply:
Yes, we drop information that is not common to these three types so their labels match.